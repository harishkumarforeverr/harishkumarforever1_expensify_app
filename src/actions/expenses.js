import { v4 as uuid } from 'uuid';
import database from "../firebase/firebase";
// ADD_EXPENSE
export const addExpense = (expense) => ({
    type: "ADD_EXPENSE",
    expense
});
//StartAddExpense
export const StartAddExpense=(expense={})=>{
    return (dispatch)=>{
          const { description = "",  note = "", amount = 0, createdAt = 0  }=expense
          const addingExpensesObject={description,note,amount,createdAt };
          return database.ref("expenses").push(addingExpensesObject).then((snapshot)=>{
              dispatch(addExpense({
                  id:snapshot.key,
                  ...addingExpensesObject
              }))
          });
    }
}

// REMOVE_EXPENSE

export const removeExpense = ({ id } = {}) => ({
    type: "REMOVE_EXPENSE",
    id
});
// EDIT_EXPENSE

export const editExpanse = (id, update) => ({
    type: "EDIT_EXPANSE",
    id,
    update
})

//SET expenses
export const setExpenses=(expense)=>({
    type:"SET_EXPENSES",
    expense 
})

export const startSetExpenses=()=>{
    return (dispatch)=>{
        return database.ref("expenses").once("value").then((snapshot)=>{   
            const expenses=[];
            snapshot.forEach((childSnapShot)=>{     
                expenses.push({
                    id:childSnapShot.key, 
                    ...childSnapShot.val()
              })
            })
            dispatch(setExpenses(expenses));
        })
    }
}