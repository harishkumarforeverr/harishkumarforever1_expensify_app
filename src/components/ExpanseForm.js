import moment from 'moment';
import React from 'react'; 
import 'react-dates/initialize';
import {SingleDatePicker} from 'react-dates';

export default class ExpanseForm extends React.Component{
    constructor(props){   
       super(props);
       this.state={ 
            description:props.expense ? props.expense.description:"",
            note:props.expense ? props.expense.note:"",
            amount:props.expense ? (props.expense.amount/100).toString() :"",
            createdAt:props.expense ? moment(props.expense.createdAt) : moment(),
            focused:false,
            formErrorMessage:""
        };
    }
    onDescriptionChange=(e)=>{ 
        this.setState(()=>{
            const description=e.target.value;
            return {
                description
            }
        })
    } 
    onNotesChange=(e)=>{
        this.setState(()=>{
            const note=e.target.value
            return {
                note
            }
        })
    }
    onAmountChange=(e)=>{ 
        const amount=e.target.value ;
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
            this.setState(()=>{
                return {
                    amount
                }
            });
        }  
    } 
    onDateChange = (createdAt)=>{
        if(createdAt){
            this.setState({createdAt})
        }
    }
    onSubmit=(e)=>{
      e.preventDefault(); 
      if(this.state.description && this.state.amount){ 
        this.setState(()=>{
            return {
                formErrorMessage:""
            }
        }); 
       this.props.onSubmit({
           description:this.state.description,
           note:this.state.note,
           amount:parseFloat(this.state.amount,10)*100,
           createdAt:this.state.createdAt.valueOf() 
       });       
      }
      else{ 
        this.setState(()=>{
            return {
                formErrorMessage:"please provide Description and Amount (mandatory.)"
            }
        }) 
        
      }
    }
    render(){
        return (   
            <form className="form" onSubmit={this.onSubmit}>
             {this.state.formErrorMessage && <h4 className="form__error"> {this.state.formErrorMessage} </h4>}
              <input 
              className="text-input"
               type="text"
               value={this.state.description}
               placeholder="add your expanses" 
               autoFocus
               onChange={this.onDescriptionChange}
               />
               <input 
               className="text-input"
               type="number" 
               placeholder="add amount" 
               value={this.state.amount}
               onChange={this.onAmountChange}
               />
               <SingleDatePicker 
               date={this.state.createdAt}
               onDateChange={this.onDateChange}
               focused={this.state.focused}
               onFocusChange={ ({focused}) =>this.setState({ focused }) }
               numberOfMonths={1}
               isOutsideRange={()=>false}
               />
               <textarea
               className="textarea"
               placeholder="add a notes to your expenses (Optional !)"
               value={this.state.note}
               onChange={this.onNotesChange}
               >

               </textarea>
               <div> 
                    <button class="button"> Save Expense </button>
               </div>
            </form> 
        );
    }
}