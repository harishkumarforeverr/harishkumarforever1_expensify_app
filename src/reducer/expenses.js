
// expanse Reducer 
const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
    switch (action.type) {  
        case "ADD_EXPENSE":
            return ([
                ...state,
                action.expense
            ]);
        case "REMOVE_EXPENSE":
            return state.filter((expense) => expense.id !== action.id);
        case "EDIT_EXPANSE" :
            return state.map((expanse)=>{
               if(expanse.id===action.id){
                   return {
                    ...expanse,
                    ...action.update
                   }
               }
               else{
                   return expanse;
               }
            });
        case "SET_EXPENSES":
            return action.expense; 
        default: return state
    }
}
