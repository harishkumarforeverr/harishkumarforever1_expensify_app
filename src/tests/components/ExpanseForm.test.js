import React from 'react';
import {shallow} from "enzyme";
import moment from "moment";
import ExpanseForm from "../../components/ExpanseForm";
import expenses from "../fixtures/expenses";
 
test("src/test/componet/ExpanseForm  :: should render component correctly",()=>{
    const wrapper=shallow(<ExpanseForm />);
    expect(wrapper).toMatchSnapshot(); 
})


test("src/test/componet/ExpanseForm  :: should render component correctly",()=>{
    const wrapper=shallow(<ExpanseForm expense={expenses[2]} />);
    expect(wrapper).toMatchSnapshot(); 
}) 

test("src/test/componet/ExpanseForm  :: checking for form submit ",()=>{
    const wrapper=shallow(<ExpanseForm />); 
    expect(wrapper).toMatchSnapshot();
    wrapper.find("form").simulate("submit",{
        preventDefault : ()=>{}
    });
    expect(wrapper.state('formErrorMessage').length).toBeGreaterThan(0);
    expect(wrapper).toMatchSnapshot();
});

test("src/test/componet/ExpanseForm  ::  should render input[0] correctly ",()=>{
     const value="harish809";
      const wrapper=shallow(<ExpanseForm />);
      wrapper.find("input").at(0).simulate("change",{
          target:{
             value
          }
      })
      expect(wrapper.state("description")).toBe(value);
});
//textarea
test("src/test/componet/ExpanseForm  ::  should render textarea correctly ",()=>{
    const value="harish809";
     const wrapper=shallow(<ExpanseForm />);
     wrapper.find("textarea").simulate("change",{
         target:{
            value
         }
     })
     expect(wrapper.state("note")).toBe(value);
});

//valid amount
test("src/test/componet/ExpanseForm  ::  should render valid amount correctly ",()=>{
    const value="123.50";
     const wrapper=shallow(<ExpanseForm />);
     wrapper.find("input").at(1).simulate("change",{
         target:{
            value
         }
     })
     expect(wrapper.state("amount")).toBe(value);
});


//Invalid amount
test("src/test/componet/ExpanseForm  ::  should render Invalid amount correctly ",()=>{
    const value="123.530";
     const wrapper=shallow(<ExpanseForm />);
     wrapper.find("input").at(1).simulate("change",{
         target:{
            value
         }
     })
     expect(wrapper.state("amount")).toBe("");
});


test("src/test/componet/ExpanseForm  ::  form must be submit with valid date and using spy in our code ",()=>{
    const onSubmitSpy=jest.fn(); 
    const wrapper=shallow(<ExpanseForm expense={expenses[2]}  onSubmit={onSubmitSpy}  />); 
    wrapper.find("form").simulate("submit",{
        preventDefault:()=>{} 
    });
    expect(wrapper.state("formErrorMessage")).toBe(""); 
    expect(onSubmitSpy).toHaveBeenLastCalledWith({  
        description:expenses[2].description,
        note:undefined,
        amount:expenses[2].amount,
        createdAt:expenses[2].createdAt
    });
})

test("src/test/componet/ExpanseForm  ::  should render SingleDatePicker onDateChange() correctly ",()=>{
     const now=moment();
     const wrapper=shallow(<ExpanseForm />);
     wrapper.find("withStyles(SingleDatePicker)").prop("onDateChange")(now);  
     expect(wrapper.state("createdAt")).toEqual(now);
});


test("src/test/componet/ExpanseForm  ::  should render SingleDatePicker onFocusChange() correctly ",()=>{
    const now=moment();
    const wrapper=shallow(<ExpanseForm />);
    wrapper.find("withStyles(SingleDatePicker)").prop("onFocusChange")({focused:true});  
    expect(wrapper.state("focused")).toEqual(true);
});

test("src/test/componet/ExpanseForm  ::  should render SingleDatePicker onFocusChange() correctly ",()=>{
    const now=moment();
    const wrapper=shallow(<ExpanseForm />);
    wrapper.find("withStyles(SingleDatePicker)").prop("onFocusChange")({focused:false});  
    expect(wrapper.state("focused")).toEqual(false);
});
