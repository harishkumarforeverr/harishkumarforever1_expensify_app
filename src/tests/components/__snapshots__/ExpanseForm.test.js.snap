// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/test/componet/ExpanseForm  :: checking for form submit  1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="add your expanses"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="add amount"
      type="number"
      value=""
    />
    <withStyles(SingleDatePicker)
      anchorDirection="left"
      appendToBody={false}
      block={false}
      calendarInfoPosition="bottom"
      customCloseIcon={null}
      customInputIcon={null}
      date={"1970-01-01T00:00:00.000Z"}
      dayPickerNavigationInlineStyles={null}
      daySize={39}
      disableScroll={false}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      horizontalMonthPadding={13}
      id="date"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepFocusOnInput={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPosition="navPositionTop"
      navPrev={null}
      noBorder={false}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsSelected": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month.",
          "jumpToPrevMonth": "Move backward to switch to the previous month.",
          "keyboardBackwardNavigationInstructions": "Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardForwardNavigationInstructions": "Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
          "moveFocusByOneMonth": "Switch months.",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
          "openThisPanel": "Open this panel.",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field.",
          "roleDescription": "datepicker",
          "selectFocusedDate": "Select the date in focus.",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      regular={false}
      renderCalendarInfo={null}
      renderDayContents={null}
      renderMonthElement={null}
      renderMonthText={null}
      renderNavNextButton={null}
      renderNavPrevButton={null}
      renderWeekHeaderElement={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      small={false}
      verticalHeight={null}
      verticalSpacing={22}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
    <textarea
      onChange={[Function]}
      placeholder="add a notes to your expenses (Optional !)"
      value=""
    />
    <button>
       Add Expense 
    </button>
  </form>
</div>
`;

exports[`src/test/componet/ExpanseForm  :: checking for form submit  2`] = `
<div>
  <h4>
     
    please provide Description and Amount (mandatory.)
     
  </h4>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="add your expanses"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="add amount"
      type="number"
      value=""
    />
    <withStyles(SingleDatePicker)
      anchorDirection="left"
      appendToBody={false}
      block={false}
      calendarInfoPosition="bottom"
      customCloseIcon={null}
      customInputIcon={null}
      date={"1970-01-01T00:00:00.000Z"}
      dayPickerNavigationInlineStyles={null}
      daySize={39}
      disableScroll={false}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      horizontalMonthPadding={13}
      id="date"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepFocusOnInput={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPosition="navPositionTop"
      navPrev={null}
      noBorder={false}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsSelected": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month.",
          "jumpToPrevMonth": "Move backward to switch to the previous month.",
          "keyboardBackwardNavigationInstructions": "Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardForwardNavigationInstructions": "Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
          "moveFocusByOneMonth": "Switch months.",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
          "openThisPanel": "Open this panel.",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field.",
          "roleDescription": "datepicker",
          "selectFocusedDate": "Select the date in focus.",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      regular={false}
      renderCalendarInfo={null}
      renderDayContents={null}
      renderMonthElement={null}
      renderMonthText={null}
      renderNavNextButton={null}
      renderNavPrevButton={null}
      renderWeekHeaderElement={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      small={false}
      verticalHeight={null}
      verticalSpacing={22}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
    <textarea
      onChange={[Function]}
      placeholder="add a notes to your expenses (Optional !)"
      value=""
    />
    <button>
       Add Expense 
    </button>
  </form>
</div>
`;

exports[`src/test/componet/ExpanseForm  :: should render component correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="add your expanses"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="add amount"
      type="number"
      value=""
    />
    <withStyles(SingleDatePicker)
      anchorDirection="left"
      appendToBody={false}
      block={false}
      calendarInfoPosition="bottom"
      customCloseIcon={null}
      customInputIcon={null}
      date={"1970-01-01T00:00:00.000Z"}
      dayPickerNavigationInlineStyles={null}
      daySize={39}
      disableScroll={false}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      horizontalMonthPadding={13}
      id="date"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepFocusOnInput={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPosition="navPositionTop"
      navPrev={null}
      noBorder={false}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsSelected": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month.",
          "jumpToPrevMonth": "Move backward to switch to the previous month.",
          "keyboardBackwardNavigationInstructions": "Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardForwardNavigationInstructions": "Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
          "moveFocusByOneMonth": "Switch months.",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
          "openThisPanel": "Open this panel.",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field.",
          "roleDescription": "datepicker",
          "selectFocusedDate": "Select the date in focus.",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      regular={false}
      renderCalendarInfo={null}
      renderDayContents={null}
      renderMonthElement={null}
      renderMonthText={null}
      renderNavNextButton={null}
      renderNavPrevButton={null}
      renderWeekHeaderElement={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      small={false}
      verticalHeight={null}
      verticalSpacing={22}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
    <textarea
      onChange={[Function]}
      placeholder="add a notes to your expenses (Optional !)"
      value=""
    />
    <button>
       Add Expense 
    </button>
  </form>
</div>
`;

exports[`src/test/componet/ExpanseForm  :: should render component correctly 2`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="add your expanses"
      type="text"
      value="creatied card"
    />
    <input
      onChange={[Function]}
      placeholder="add amount"
      type="number"
      value="80"
    />
    <withStyles(SingleDatePicker)
      anchorDirection="left"
      appendToBody={false}
      block={false}
      calendarInfoPosition="bottom"
      customCloseIcon={null}
      customInputIcon={null}
      date={"1970-01-05T00:00:00.000Z"}
      dayPickerNavigationInlineStyles={null}
      daySize={39}
      disableScroll={false}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      horizontalMonthPadding={13}
      id="date"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepFocusOnInput={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPosition="navPositionTop"
      navPrev={null}
      noBorder={false}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsSelected": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month.",
          "jumpToPrevMonth": "Move backward to switch to the previous month.",
          "keyboardBackwardNavigationInstructions": "Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardForwardNavigationInstructions": "Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
          "moveFocusByOneMonth": "Switch months.",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
          "openThisPanel": "Open this panel.",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field.",
          "roleDescription": "datepicker",
          "selectFocusedDate": "Select the date in focus.",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      regular={false}
      renderCalendarInfo={null}
      renderDayContents={null}
      renderMonthElement={null}
      renderMonthText={null}
      renderNavNextButton={null}
      renderNavPrevButton={null}
      renderWeekHeaderElement={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      small={false}
      verticalHeight={null}
      verticalSpacing={22}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
    <textarea
      onChange={[Function]}
      placeholder="add a notes to your expenses (Optional !)"
    />
    <button>
       Add Expense 
    </button>
  </form>
</div>
`;
